core/src/main/java/org/apache/accumulo/core/util/Halt.java

```diff
@@ -30,25 +30,19 @@ public class Halt {
 
   public static void halt(final String msg) {
     // ACCUMULO-3651 Changed level to error and added FATAL to message for slf4j compatibility
-    halt(0, new Runnable() {
-      @Override
-      public void run() {
-        log.error("FATAL {}", msg);
-      }
-    });
+    halt(0, "FATAL " + msg, null);
   }
 
   public static void halt(final String msg, int status) {
-    halt(status, new Runnable() {
-      @Override
-      public void run() {
-        log.error("FATAL {}", msg);
-      }
-    });
+    halt(status, "FATAL " + msg, null);
   }
 
-  public static void halt(final int status, Runnable runnable) {
+  public static void halt(final int status, String msg, Runnable runnable) {
 
+    // Printing to stderr and to the log in case the message does not make
+    // it to the log. This could happen if an asynchronous logging impl is used
+    System.err.println(msg);
+    log.error(msg);
```

Comment:
These should be moved into the try/finally.

Reply:
You mean at the beginning of the try block? I don't think it should go into the finally block, that would be too late.